### ISSUE TITLE
Produced a missed block 

### ISSUE THREAD

**Issue by PacoBits** [Consensys-Incorporated/teku#6658]
Our setup has produced a missed block https://beaconcha.in/slot/5501765
We have been producing blocks correctly and after that incident it is still producing blocks  correctly.
We have a 3-layer configuration with the validator client, the Beacon node and the execution client running separately.
We are using Mev-boost

This issue its RARE and not replicable at this moment and seems related to a reorg chain. 

Validator: 326216
Slot: 5501765
Epoch: 171930
Proposed block should be: 16334246

These are our versions and fee recipent setup:

Validator set up and: version: Lighthouse/v3.3.0-bf533c8
         lighthouse vc
         --beacon-nodes http://ip1,http://ip2,http://ip3
         --network mainnet
         --metrics
         --metrics-allow-origin *
         --graffiti "Block by xxxxxxx"
         --suggested-fee-recipient 0x388Cxxxxxxxxxxxx
         --init-slashing-protection
         --metrics-port xxx
         --http                           
         --builder-proposals

Beacon set up - teku/v22.12.0/linux-x86_64/-eclipseadoptium-openjdk64bitservervm-java-17
   command: ["--network=mainnet",
              "--data-base-path=/opt/teku/data",
              "--eth1-endpoints=http://localhost:8545",
              "--p2p-port=9000",
              "--rest-api-enabled=true",
              "--rest-api-host-allowlist=*",
              "--metrics-enabled=true",
              "--p2p-subscribe-all-subnets-enabled=true",
              "--ee-endpoint=http://127.0.0.1:aaaa",
              "--ee-jwt-secret-file=xxxxxxxxxxxx",
              "--validators-graffiti='Block by Stakely.io - btk01'",
              "--validators-proposer-default-fee-recipient=0x718xxxxx",
              "--validators-builder-registration-default-enabled=true",
              "--validators-proposer-blinded-blocks-enabled=true",
              "--builder-endpoint=http://localhost:18550",
              "--log-destination=CONSOLE",
              "--logging=INFO"

Execution Client - git_branch=heads/v2.32.0 git_tag=v2.32.0-dirty git_commit=021891a311430ffefde77fcceda567ba324e9abe
     erigon
     --prune htc
     --prune.r.before=11184524     
     --maxpeers 100
     --nat any
     --http
     --http.addr 0.0.0.0
     --http.port 8545
     --http.api=eth,debug,net,trace,web3,erigon,engine,txpool
     --authrpc.port aaaa
     --authrpc.addr xxxxx
     --authrpc.jwtsecret xxxxx
     --private.api.addr=localhost:9091
     --chain mainnet
     --metrics
     --port 30303
     --torrent.port 42070
     --datadir=/home/erigon/.local/share/erigon
     --authrpc.vhosts "xxxxxxxxxxxx"
     --externalcl
     --port 30302

I attached logs from the time the block produced.
[teku_04012023.log](https://github.com/ConsenSys/teku/files/10347073/teku_04012023.log)
[erigon_04012023.log](https://github.com/ConsenSys/teku/files/10347074/erigon_04012023.log)
[mev-04012023.log](https://github.com/ConsenSys/teku/files/10347075/mev-04012023.log)
[LighthouseVC_04012023.log](https://github.com/ConsenSys/teku/files/10347076/LighthouseVC_04012023.log)

We suspect that the problem is due to the fact that the block was generated during a chain reorganization.

**Comment by benjaminion** [comment]
Some preliminary observations from the logs.

 - Your block for slot 5501765 should have been published at 15:13:23 (or within 1 or 2 seconds after that).
 - According to your Teku logs, builder bids were received at 15:13:30, and the execution payload at 15:13:36.
 - Assuming that Teku published the block immediately then it is already 13s late.
 - This is confirmed by my node, which received your block 14.2 seconds late.
 - As a result of its lateness, it was reorged out (orphaned). My node initially made your block its head, but received 5501766 only 1.3 seconds after it which built on 5501764 (since its proposer had not seen your block). Therefore your block was orphaned. This is the reorg you see in Teku's log: your block is `449b9d74...`, but it gets superseded by `093503e8...`

> We suspect that the problem is due to the fact that the block was generated during a chain reorganization.

As above, I think its lateness actually caused the reorg.

Note that the Lighthouse logs have `Jan 04 15:13:38.549 INFO Successfully published block            slot: 5501765`, which is  0.8s _after_ my node saw the block, so there looks to be some delay in LH's logging. But I'm pretty sure this is not the root cause, since the publishing window is long-gone by this point. LH's attestations are mostly being signed on time, though there looks to be a delay dealing with one of them (at 15:13:30.768) which might indicate performance trouble.

Also, I don't know what this LH message means, `Jan 04 15:13:19.481 ERRO Unable to publish proposer preparation to all beacon nodes`. And again at `Jan 04 15:13:35.311`, just ahead of the successful publishing. It's worth investigating as there's a chance it's related

So, the question is, why did your node publish the block so late? Looking at the MEV logs, they only start receiving bids at 15:13:30, which is already 7s after the start of the slot, and `getPayload` is not called until 6s later, together accounting for the 13s lateness in publishing of the block.

In short:

 - MEV boost should have been receiving quotes from the relays much sooner. Are there any lines before the ones in _mev-04012023.log_, specifically from 15:13:20 onwards?
 - And it should have received the payload much sooner too. Was there a short network outage or degradation at the time? How was the CPU load on the MEV boost machine?

**Comment by cmeisl** [comment]
Blocknative received 2 getPayload requests for this slot. Both requests were serviced in about 100ms (including firewall and load balancer processing).

The first `getPayload` was requested at `2023-01-04T15:13:36.745173342Z`
The second `getPayload` was requested at `2023-01-04T15:13:37.930028982Z`

From the mev-boost log above, it looks like payload was received in less than 1 second (resolution of log timestamps).

Blocknative engineering is curious why we received 2 `getpPayload` requests. Were there 2 instances of mev-boost running?

**Comment by PacoBits** [comment]
Good analysis @benjaminion.
But it is weird that erigon does produce a first block 16334246 at 15:13:13. Shouldn't teku have used that block if mev takes too long to respond? 
Although it is true that teku returns to produce the block at 15:13:30, maybe it is due to a disconnection between VC and CL during the generation of the first block at 15:13:13.

```
erigon-execution-client     | [INFO] [01-04|15:13:13.155] [2/16 Headers] Waiting for Consensus Layer...
erigon-execution-client     | [INFO] [01-04|15:13:13.155] Building block...
erigon-execution-client     | [INFO] [01-04|15:13:13.158] [1/5 MiningCreateBlock] Start mine       block=16334246 baseFee=26337366647 gasLimit=30000000
erigon-execution-client     | [INFO] [01-04|15:13:13.179] [2/5 MiningExecution] Mined transactions count=41 gas_used=4744583
erigon-execution-client     | [INFO] [01-04|15:13:13.204] Built block                              hash=0xf5fd082af0da308143c235d0aee05fefde8971aec0f5f408582d3c01aecb5e29 height=16334246 txs=41 gas used %=15.815 time=49.010176ms
erigon-execution-client     | [INFO] [01-04|15:13:30.487] Building block...
erigon-execution-client     | [INFO] [01-04|15:13:30.490] [1/5 MiningCreateBlock] Start mine       block=16334246 baseFee=26337366647 gasLimit=30000000
erigon-execution-client     | [INFO] [01-04|15:13:30.517] Received GetPayloadV1                    payloadId=2104
erigon-execution-client     | [INFO] [01-04|15:13:30.565] [2/5 MiningExecution] Mined transactions count=220 gas_used=20957295
erigon-execution-client     | [INFO] [01-04|15:13:30.690] Built block                              hash=0x15e5e33812d4e8186679d5eed27bd41da32e9940ede92c2a201a0d6539b8077a height=16334246 txs=220 gas used %=69.858 time=202.351067ms
```

**Comment by PacoBits** [comment]
@cmeisl  I can't find in our logs two different GetPayload request. We have 2 instances of mev-boost, one for each CL. I paste here the logs of both.

Are the two Getpayload requests identical or does the blockhash change?

Erigon Teku MEv-Boost log --> Main CL node
```

mev-boost  | time="2023-01-04T15:12:47Z" level=info msg="http: GET /eth/v1/builder/status 200" duration=0.000032 method=GET path=/eth/v1/builder/status status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:12:59Z" level=info msg="http: GET /eth/v1/builder/status 200" duration=0.000034 method=GET path=/eth/v1/builder/status status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:13:11Z" level=info msg="http: GET /eth/v1/builder/status 200" duration=0.000027 method=GET path=/eth/v1/builder/status status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:13:23Z" level=info msg="http: GET /eth/v1/builder/status 200" duration=0.000081 method=GET path=/eth/v1/builder/status status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:13:30Z" level=debug msg=getHeader method=getHeader parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c pubkey=0x897c920ef6325f43ae60ee502a5145db9fac762c9014fcc5557ff6c4efdf7d8bba6c88c82e3909e6058b458ab5137767 slot=5501765 >mev-boost  | time="2023-01-04T15:13:30Z" level=debug msg="bid received" blockHash=0x65e2a3cdf689a5e50df12e9cef17bbe41befaeca2a9184444b9fb58e4996baba blockNumber=16334246 method=getHeader parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c pubkey=0x897c920>mev-boost  | time="2023-01-04T15:13:30Z" level=debug msg="new best bid" blockHash=0x65e2a3cdf689a5e50df12e9cef17bbe41befaeca2a9184444b9fb58e4996baba blockNumber=16334246 method=getHeader parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c pubkey=0x897c920>mev-boost  | time="2023-01-04T15:13:30Z" level=debug msg="bid received" blockHash=0x2f3e59e860d1474b6a5447c1c88562637f642dea18b8f0aa1d2aadfb99470733 blockNumber=16334246 method=getHeader parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c pubkey=0x897c920>mev-boost  | time="2023-01-04T15:13:30Z" level=debug msg="new best bid" blockHash=0x2f3e59e860d1474b6a5447c1c88562637f642dea18b8f0aa1d2aadfb99470733 blockNumber=16334246 method=getHeader parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c pubkey=0x897c920>mev-boost  | time="2023-01-04T15:13:30Z" level=debug msg="bid received" blockHash=0x00c83fc9eb8adf0e710d94a5391af4f5e643a398f8560d83007f5f4d324eeab4 blockNumber=16334246 method=getHeader parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c pubkey=0x897c920>mev-boost  | time="2023-01-04T15:13:30Z" level=debug msg="new best bid" blockHash=0x00c83fc9eb8adf0e710d94a5391af4f5e643a398f8560d83007f5f4d324eeab4 blockNumber=16334246 method=getHeader parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c pubkey=0x897c920>mev-boost  | time="2023-01-04T15:13:30Z" level=debug msg="bid received" blockHash=0xda1737744d040be74aade4ea94adde4329c9b53ab74f8a7be0971c6119cab2ae blockNumber=16334246 method=getHeader parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c pubkey=0x897c920>mev-boost  | time="2023-01-04T15:13:30Z" level=debug msg="bid received" blockHash=0x74b6869f571e8e82786050cb74872936286d0c33f1448e2988c209854bec142a blockNumber=16334246 method=getHeader parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c pubkey=0x897c920>mev-boost  | time="2023-01-04T15:13:30Z" level=debug msg="no-content response" method=getHeader parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c pubkey=0x897c920ef6325f43ae60ee502a5145db9fac762c9014fcc5557ff6c4efdf7d8bba6c88c82e3909e6058b458ab5137767 s>mev-boost  | time="2023-01-04T15:13:30Z" level=info msg="best bid" blockHash=0x00c83fc9eb8adf0e710d94a5391af4f5e643a398f8560d83007f5f4d324eeab4 blockNumber=16334246 method=getHeader parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c pubkey=0x897c920ef632>mev-boost  | time="2023-01-04T15:13:30Z" level=info msg="http: GET /eth/v1/builder/header/5501765/0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c/0x897c920ef6325f43ae60ee502a5145db9fac762c9014fcc5557ff6c4efdf7d8bba6c88c82e3909e6058b458ab5137767 200" duration=0.2>
mev-boost  | time="2023-01-04T15:13:35Z" level=info msg="http: GET /eth/v1/builder/status 200" duration=0.000035 method=GET path=/eth/v1/builder/status status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:13:36Z" level=debug msg=getPayload method=getPayload version=v1.4.0
mev-boost  | time="2023-01-04T15:13:36Z" level=debug msg="calling getPayload" blockHash=0x00c83fc9eb8adf0e710d94a5391af4f5e643a398f8560d83007f5f4d324eeab4 method=getPayload parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c slot=5501765 url="https://buil>mev-boost  | time="2023-01-04T15:13:36Z" level=info msg="received payload from relay" blockHash=0x00c83fc9eb8adf0e710d94a5391af4f5e643a398f8560d83007f5f4d324eeab4 method=getPayload parentHash=0x4c98b9fe68d8a926506170f37d023f27d783ce903aaef234234ff0cd4dc4c51c slot=5501765 url="http>
mev-boost  | time="2023-01-04T15:13:36Z" level=info msg="http: POST /eth/v1/builder/blinded_blocks 200" duration=0.498549 method=POST path=/eth/v1/builder/blinded_blocks status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:13:47Z" level=info msg="http: GET /eth/v1/builder/status 200" duration=0.000028 method=GET path=/eth/v1/builder/status status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:13:59Z" level=info msg="http: GET /eth/v1/builder/status 200" duration=0.000032 method=GET path=/eth/v1/builder/status status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:14:11Z" level=info msg="http: GET /eth/v1/builder/status 200" duration=0.000057 method=GET path=/eth/v1/builder/status status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:14:23Z" level=info msg="http: GET /eth/v1/builder/status 200" duration=0.000060 method=GET path=/eth/v1/builder/status status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:14:35Z" level=info msg="http: GET /eth/v1/builder/status 200" duration=0.000064 method=GET path=/eth/v1/builder/status status=200 version=v1.4.0
```

Lighthouse  GEth Mev-Boost log --> Bakup CL node
```

mev-boost  | time="2023-01-04T15:12:23Z" level=info msg="http: POST /eth/v1/builder/validators 200" duration=0.481224 method=POST path=/eth/v1/builder/validators status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:12:23Z" level=info msg="http: POST /eth/v1/builder/validators 200" duration=0.478543 method=POST path=/eth/v1/builder/validators status=200 version=v1.4.0
mev-boost  | time="2023-01-04T15:12:26Z" level=warning msg="error calling registerValidator on relay" error="HTTP error response: 502 / \n<html><head>\n<meta http-equiv=\"content-type\" content=\"text/html;charset=utf-8\">\n<title>502 Server Error</title>\n</head>\n<body text=#000>mev-boost  | time="2023-01-04T15:12:26Z" level=warning msg="error calling registerValidator on relay" error="HTTP error response: 502 / \n<html><head>\n<meta http-equiv=\"content-type\" content=\"text/html;charset=utf-8\">\n<title>502 Server Error</title>\n</head>\n<body text=#000>mev-boost  | time="2023-01-04T15:15:11Z" level=debug msg=getHeader method=getHeader parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 pubkey=0xa33d785261aac70a955349729221103983f89a4a8cc57cfbc9dfdbb8582b85f55c604a854cbe8d3da1cb326db7a665fd slot=5501774 >mev-boost  | time="2023-01-04T15:15:11Z" level=debug msg="bid received" blockHash=0x8619c41241c58199c2171cf1e49ea1745d21a852d040c5661c50af91dadc19a1 blockNumber=16334254 method=getHeader parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 pubkey=0xa33d785>mev-boost  | time="2023-01-04T15:15:11Z" level=debug msg="new best bid" blockHash=0x8619c41241c58199c2171cf1e49ea1745d21a852d040c5661c50af91dadc19a1 blockNumber=16334254 method=getHeader parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 pubkey=0xa33d785>mev-boost  | time="2023-01-04T15:15:11Z" level=debug msg="bid received" blockHash=0x829a2d9ec16594a7675185b1ec14ac300034284623f8d18e3b9e3991c1425539 blockNumber=16334254 method=getHeader parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 pubkey=0xa33d785>mev-boost  | time="2023-01-04T15:15:11Z" level=debug msg="new best bid" blockHash=0x829a2d9ec16594a7675185b1ec14ac300034284623f8d18e3b9e3991c1425539 blockNumber=16334254 method=getHeader parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 pubkey=0xa33d785>mev-boost  | time="2023-01-04T15:15:11Z" level=debug msg="bid received" blockHash=0x5f27f2463a168a1d31a25c418a80bc206a3cad3662025d4c2950fbb51f844b43 blockNumber=16334254 method=getHeader parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 pubkey=0xa33d785>mev-boost  | time="2023-01-04T15:15:11Z" level=debug msg="bid received" blockHash=0xafb09d1e05e6dcc0edac183610c9910e6d183eefada0b7e5d4902ee1bac4f783 blockNumber=16334254 method=getHeader parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 pubkey=0xa33d785>mev-boost  | time="2023-01-04T15:15:11Z" level=debug msg="bid received" blockHash=0xbe100c2e814f72adac5c623d854ee73fbe4f372ec240bca94ed767458ff30019 blockNumber=16334254 method=getHeader parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 pubkey=0xa33d785>mev-boost  | time="2023-01-04T15:15:11Z" level=debug msg="no-content response" method=getHeader parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 pubkey=0xa33d785261aac70a955349729221103983f89a4a8cc57cfbc9dfdbb8582b85f55c604a854cbe8d3da1cb326db7a665fd s>mev-boost  | time="2023-01-04T15:15:11Z" level=info msg="best bid" blockHash=0x829a2d9ec16594a7675185b1ec14ac300034284623f8d18e3b9e3991c1425539 blockNumber=16334254 method=getHeader parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 pubkey=0xa33d785261aa>mev-boost  | time="2023-01-04T15:15:11Z" level=info msg="http: GET /eth/v1/builder/header/5501774/0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803/0xa33d785261aac70a955349729221103983f89a4a8cc57cfbc9dfdbb8582b85f55c604a854cbe8d3da1cb326db7a665fd 200" duration=0.3>mev-boost  | time="2023-01-04T15:15:11Z" level=debug msg=getPayload method=getPayload version=v1.4.0
mev-boost  | time="2023-01-04T15:15:11Z" level=debug msg="calling getPayload" blockHash=0x829a2d9ec16594a7675185b1ec14ac300034284623f8d18e3b9e3991c1425539 method=getPayload parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 slot=5501774 url="https://boos>mev-boost  | time="2023-01-04T15:15:12Z" level=info msg="received payload from relay" blockHash=0x829a2d9ec16594a7675185b1ec14ac300034284623f8d18e3b9e3991c1425539 method=getPayload parentHash=0xaaf3bfeabab54a6a77af0488a81571badd9fbf2dcb204c6162a16b34d2882803 slot=5501774 url="http>mev-boost  | time="2023-01-04T15:15:12Z" level=info msg="http: POST /eth/v1/builder/blinded_blocks 200" duration=0.535318 method=POST path=/eth/v1/builder/blinded_blocks status=200 version=v1.4.0
```

**Comment by PacoBits** [comment]
I have tried to sort in time the most relevant information events of each component  :
One of the CL is disconnected, maybe teku.
`VC_LIGHTHOUSE         | Jan 04 15:13:05.000 INFO Connected to beacon node(s)             synced: 2, available: 2, total: 3, service: notifier`


Two CLs have a connection problem with VC. CLS-->  Teku and LH . Logs shows their IPs
`VC_LIGHTHOUSE         | Jan 04 15:13:12.202 ERRO Failed to subscribe validators     
`


The service is working well 5 seconds later. 
`VC_LIGHTHOUSE         | Jan 04 15:13:17.000 INFO Connected to beacon node(s) 
`
We have not much information about teku logs, but erigon built first block at 15:13:13

I attach the file extracting logs .
[missed_proposal_relevant04012023.log](https://github.com/ConsenSys/teku/files/10350523/missed_proposal_relevant04012023.log)

**Comment by StefanBratanov** [comment]
My observation from the logs:

- I don't see issues with mev-boost. CL asked for a bid and received it quick enough. Same goes for the execution payload.
- There is 6s delay between receiving the bid and asking for the execution payload. In the middle of it, there is `Completed regeneration of block...` log message which could be related to the delay.
- I am not sure whether the 7s delay between start of the slot and asking the builder for a bid is coming from Teku or the VC since there is no log of when VC requested Teku for a block. Will check Teku if `Completed regeneration of block...` could be related to this delay as well.
- I am concerned about this timeout log against the Teku BN: `Jan 04 15:13:19.481 ERRO Unable to publish proposer preparation to all beacon nodes`. It is possible there were some network problems at the time, which could explain the initial delay maybe.

**Comment by cmeisl** [comment]
@PacoBits The 2 `getPayload` requests were for the same blockhash with the same pubkey. This is not an issue, we were just curious about it.

Do I understand correctly that you are running 2 separate validators "in parallel" - i.e. they have same pubkey registration and therefore both attempt to propose for the same slot?

**Comment by PacoBits** [comment]
@cmeisl  about running in parallel is a bit more complex
We are using a remote signer and running a VC connected to several CLs, but the VC only uses the first (primary) CL. In case the primary CL goes down, the VC will connect to the second CL (bakup) and if this also fails, the new CL will connect to the third and last backup CL. In those cases the pubkey wolud be register in several CL.
Note Web3signer avoid slashing situations. So in this particular case, we can see in VC logs a connection problem between VC and the primary CL and the backup CL, so the same pubkey was registered in almost two CLs, the primary and the second backup CL. 

Regarding this particular problem, with mev-boost working fine, machine resources at good state,  and with the evidence evidence of a one-time network problem (VC logs), it may not be possible to avoid the missed proposal, since it was initiated when the network problem occurred.

@StefanBratanov  I will try to add to this thread to Lighthouse team, perhaps they can add more information about` Jan 04 15:13:19.481 ERRO Unable to publish proposer preparation to all beacon nodes`

**Comment by paulhauner** [comment]
> Also, I don't know what this LH message means, Jan 04 15:13:19.481 ERRO Unable to publish proposer preparation to all beacon nodes.

This indicates a failure to call the [`/eth/v1/validator/prepare_beacon_proposer`](https://ethereum.github.io/beacon-APIs/#/ValidatorRequiredApi/prepareBeaconProposer) endpoint. Unfortunately the log message is truncated so I can't see *why* it failed (e.g., the HTTP status code).

Publishing proposer preparation data happens in a separate process in the VC, so it shouldn't impact block production *inside the VC*. However, I could see that failing to prepare for a proposal could impact block production for the BN/EL. The Teku team seem to be doing a good job at looking into this process already, though.

**Comment by paulhauner** [comment]
> Note that the Lighthouse logs have Jan 04 15:13:38.549 INFO Successfully published block            slot: 5501765, which is 0.8s after my node saw the block, so there looks to be some delay in LH's logging. 

The process of a VC publishing a block looks something like this:

1. VC sends block to BN (HTTP call starts)
1. BN does gossip verification, forwards to the network
1. BN fully verifies and imports block
1. BN responds to HTTP call (HTTP call ends)
1. VC emits log indicating block was published.

So the delay between a remote node seeing the block and the VC logging success is likely due to the lag introduced from steps 3 & 4.

**Comment by paulhauner** [comment]
>  I am not sure whether the 7s delay between start of the slot and asking the builder for a bid is coming from Teku or the VC since there is no log of when VC requested Teku for a block.

It would indeed be interesting to know when the VC first initiated block production. We only have the VC logs for when block production is *finished*. Historically we've used the BN HTTP server logs to determine when the VC first requested a block, so our VC logs are lacking in this area. Perhaps Teku has some logging of incoming HTTP requests? It's not useful for this scenario, but it may be prudent for Lighthouse to add more logging in this area (issue: https://github.com/sigp/lighthouse/issues/3853).

Given the relatively small amount of info in the Lighthouse VC logs, I don't think there's a whole lot more I can add here sorry. I'll leave it to the capable Teku team.

**Comment by PacoBits** [comment]
@paulhauner  About:
> It would indeed be interesting to know when the VC first initiated block production.
Maybe erigon logs can help
```
erigon-execution-client     | [INFO] [01-04|15:13:13.158] [1/5 MiningCreateBlock] Start mine       block=16334246 baseFee=26337366647 gasLimit=30000000
erigon-execution-client     | [INFO] [01-04|15:13:13.179] [2/5 MiningExecution] Mined transactions count=41 gas_used=4744583
erigon-execution-client     | [INFO] [01-04|15:13:13.204] Built block                              hash=0xf5fd082af0da308143c235d0aee05fefde8971aec0f5f408582d3c01aecb5e29 height=16334246 txs=41 gas used %=15.815 time=49.010176ms
```

I would like to add the complete VC log no truncate. We can see a timeout (network issue)

[LighthouseVC_04012023_No_truncated.log](https://github.com/ConsenSys/teku/files/10358836/LighthouseVC_04012023_No_truncated.log)

**Comment by StefanBratanov** [comment]
Hi, I did a little further investigation and find it hard to pinpoint the issue. It is possible these delays happened due to a garbage collection taking place at that time or other resource issue. The timeout 3 seconds before the slot time could indicate such issue as well. 

Couple of questions:
- What heap space do you assign to Teku (-Xmx)? 5gb works well for our validators.
- Since metrics are enabled, are you using the Teku dashboard by any chance? There is a GC time graph, which is useful and should be under 5% at all times (ideally under 2%)

**Comment by PacoBits** [comment]
Hi, 

- The head space is   - "JAVA_OPTS=-Xmx4g". 
- We are not using Teku Dashboard but we can install it. We have not made any changes to the machine so the values should remain.

Actions:
- We will install the Teku dashboard and post more information about GC time here. 
- We do not modify the head space so that it does not affect the GC values.

**Comment by PacoBits** [comment]
Hi @StefanBratanov  we can see on teku dashboard several high peaks about GC. 
Our average GC values for the last 12 hours:

- G1 Old Generation     Max: 7.85%	avg: 0.309%	current: 0%
- G1 Young Generation Max: 5.18%	avg: 1.57%	current: 1.41%

Maybe we should set heap space to 5 GB and study the GC values. We are running a dockerize solution.

**Comment by StefanBratanov** [comment]
> Hi @StefanBratanov we can see on teku dashboard several high peaks about GC. Our average GC values for the last 12 hours:
> 
> * G1 Old Generation     Max: 7.85%	avg: 0.309%	current: 0%
> * G1 Young Generation Max: 5.18%	avg: 1.57%	current: 1.41%
> 
> Maybe we should set heap space to 5 GB and study the GC values. We are running a dockerize solution.

Hi @PacoBits these are  high max values. It could lead to the JVM blocking which is very likely what happened. I just noticed also you are subscribing to all subnets, which could also affect performance slightly. We are running with 5GB heap space on nodes which subscribe to all subnets as well and don't see issues so it's worth a try to increase the value as you suggested and see how that affects the GC values.

**Comment by PacoBits** [comment]
![image](https://user-images.githubusercontent.com/38920606/211739035-eeda463e-c7f8-4081-b6fe-c1195bd656af.png)
Hi @StefanBratanov we can see a significant performance improvement in GC after switching to 5GB. Maybe the problem was a punctual issue due to a GC spike at a very specific time.

**Comment by StefanBratanov** [comment]
> ![image](https://user-images.githubusercontent.com/38920606/211739035-eeda463e-c7f8-4081-b6fe-c1195bd656af.png) Hi @StefanBratanov we can see a significant performance improvement in GC after switching to 5GB. Maybe the problem was a punctual issue due to a GC spike at a very specific time.

Hi @PacoBits it does look better and I suspect that this was the issue. It can explain the JVM being blocked. In the meantime, I raised an issue to log the start of block production for Teku to help for future debugging.

**Comment by PacoBits** [comment]
Hi @StefanBratanov thank you all for your support. 

I'm keeping the issue open for 24 hours in case someone adds something and I'll close it.
